### PR TITLE
[Vincenzo Spinello] Sequence_task_1

### PR DESCRIPTION
**Pull request by vspinell**

### CODE REVIEW

File: sequence_task_1/student/include/impl.h [hunk 1]
```diff
@@ -9,32 +9,61 @@ class StaticPlaylist
 public:
     /** @todo Member traits */
 
+	using value_type = Song_t;
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: _you can also refer to container value types here_

File: sequence_task_1/student/include/impl.h [hunk 2]
```diff
@@ -9,32 +9,61 @@ class StaticPlaylist
...
+	using reference = Song_t &;
+	using const_reference = const Song_t &;
+	using iterator = typename Container::iterator;
+	using const_iterator = typename Container::const_iterator;
+	using difference_type = ptrdiff_t;
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: not needed I think, reuse container's members

File: sequence_task_1/student/include/impl.h [hunk 3]
```diff
@@ -9,32 +9,61 @@ class StaticPlaylist
...
+	StaticPlaylist& operator=(const reversable_sequence_container &other) {
+		m_tracklist.assign(other.rbegin(), other.rend());
+		return *this;
+	}
+
     /** @todo Add track from initializer */
     template<class... Args>
-    const Song_t& play(Args&&... songData);
+    const Song_t& play(Args&&... songData) {
+		return *m_tracklist.emplace(end(), std::forward<Args>(songData)...);
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: emplace_back?

File: sequence_task_1/student/include/impl.h [hunk 4]
```diff
@@ -9,32 +9,61 @@ class StaticPlaylist
...
+	}
 
     /** @todo Add track */
-    const Song_t& play(const Song_t& song);
+    const Song_t& play(const Song_t& song) { return *m_tracklist.insert(m_tracklist.end(), song); }
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: push_back?

File: sequence_task_1/student/include/impl.h [hunk 5]
```diff
@@ -9,32 +9,65 @@ class StaticPlaylist
...
+	const_iterator begin() const {return m_tracklist.cbegin();}
+	const_iterator end() const {return m_tracklist.cend();}
+
+
     StaticPlaylist() = default;
 
     /** @todo Constructor from any reversible sequence container */
 
+	template <typename reversable_sequence_container>
+	StaticPlaylist(reversable_sequence_container init) {
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: reversable_sequence_container will be the container itself, so here you're doing the copy - please don't, add the const reference!

File: sequence_task_1/student/include/impl.h [hunk 6]
```diff
@@ -9,32 +9,65 @@ class StaticPlaylist
...
+		m_tracklist.assign(init.rbegin(), init.rend());
```
Comment (smay1613) [inline sequence_task_1/student/include/impl.h]: we can use constructor

**Comment by vspinell** [comment]
Are there any changes to be applied to this task?

**Comment by smay1613** [comment]
No, everything is ok

**Pull request closed without merging**